t/lib/NAR/Helper.pm

```diff
@@ -130,6 +130,9 @@ sub connect {
         ssl_init        => $self->{ssl_init},
         vhost           => $self->{vhost},
     };
+	my $client_properties = {
```

Comment:
Since this is supposed to be optional, can you take it out of the central location, and just add tests for the optional functionality? We need to make sure this change doesn't have any additional side-effects.